mockgen/parse.go

```diff
@@ -228,17 +246,36 @@ func (p *fileParser) parseFile(importPath string, file *ast.File) (*model.Packag
 
 	var is []*model.Interface
 	for ni := range iterInterfaces(file) {
-		if _, ok := p.excludeNamesSet[ni.name.String()]; ok {
+		name := ni.name.String()
+
+		if _, ok := p.excludeNamesSet[name]; ok {
 			continue
 		}
-		i, err := p.parseInterface(ni.name.String(), importPath, ni)
+
+		if len(p.includeNamesSet) > 0 {
+			if _, ok := p.includeNamesSet[name]; !ok {
+				continue
+			}
+		}
+
+		i, err := p.parseInterface(name, importPath, ni)
 		if errors.Is(err, errConstraintInterface) {
 			continue
 		}
 		if err != nil {
 			return nil, err
 		}
 		is = append(is, i)
+
+		delete(p.includeNamesSet, name)
+	}
+	if len(p.includeNamesSet) > 0 {
+		missing := make([]string, 0, len(p.includeNamesSet))
+		for n := range p.includeNamesSet {
+			missing = append(missing, n)
+		}
+		sort.Strings(missing)
```

Comment:
We can shorten this:
```suggestion
		missing := slices.Sorted(maps.Keys))
```

(assuming import slices and maps.)